### PR TITLE
Implement air vent pressure lockout override

### PR DESCRIPTION
**Pull request by MjrLandWhale**
## About the PR

Adds a "Pressure Lockout Override" functionality to all air vents. Allowing atmos techs to override the lockout by setting an air alarm to fill.

## Why / Balance

Caused by #28370. Based on discussion in PR, this was a reasonable solution to there being no consistent way of repressurizing a spaced room.

## Technical details

- Adds PressureLockoutOverride variable to GasVentPumpComopnent
- GasVentPumpData presets now include PressureLockoutOverride
- Adds conditions in GasVentPumpSystem for the lockout override, only when we're already checking for the lockout value.

## Media

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

N/A
**Changelog**

- add: Under-pressure lock-out on air vents can be overridden with the air alarm "Fill" mode. Use it to re-pressurize spaced rooms.

### CODE REVIEW

File: Content.Server/Atmos/Piping/Unary/Components/GasVentPumpComponent.cs [hunk 1]
```diff
@@ -135,6 +135,10 @@ public float InternalPressureBound
         [ViewVariables(VVAccess.ReadWrite)]
         [DataField("depressurizePressure")]
         public float DepressurizePressure = 0;
+
+        [ViewVariables(VVAccess.ReadWrite)]
```
Comment (deltanedas) [inline Content.Server/Atmos/Piping/Unary/Components/GasVentPumpComponent.cs:139]: replace these 2 lines with just [DataField]
also false is the default bool so you dont have to specify it

**Comment by Partmedia** [comment]
Hi, thanks for the quick PR. This may be affected by the current feature freeze, so I'll at minimum wait for some feedback on whether or not this would be covered or not.

Looks good conceptually and code-wise. Were you able to give this an in-game test?

**Review by deltanedas — CHANGES_REQUESTED** [review]
i also think this should be a setting under the air vent like how widenet is for scrubbers

**Comment by MjrLandWhale** [comment]
No problem. We'll see what they say. As for testing, I validated it small scale on dev, and large scale by spacing all of medbay on origin. Lockout functions under 60kpa, unless set to fill. This change WOULD allow someone to vent distro by setting a still spaced room to fill, however that's a gameplay consequence of their own making.

**Comment by deltanedas** [comment]
someone doing that can also just put a pump and spaced passive vent on distro to empty it, which requires no access or insuls

File: Content.Server/Atmos/Piping/Unary/Components/GasVentPumpComponent.cs (hunk 1 above)
Comment (MjrLandWhale) [reply Content.Server/Atmos/Piping/Unary/Components/GasVentPumpComponent.cs]: Done. Also didn't realize that about bools, good to know. I do want to leave it defined though, just for clarity.

File: Content.Server/Atmos/Piping/Unary/Components/GasVentPumpComponent.cs [hunk 2]
```diff
@@ -135,6 +135,9 @@ public float InternalPressureBound
         [ViewVariables(VVAccess.ReadWrite)]
         [DataField("depressurizePressure")]
         public float DepressurizePressure = 0;
+
+        [DataField("pressureLockoutOverride")]
```
Comment (deltanedas) [inline Content.Server/Atmos/Piping/Unary/Components/GasVentPumpComponent.cs:139]: still dont need to specify the tag as its just camelCased variable name, literally `[DataField]` nothing else

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by MjrLandWhale** [comment]
> someone doing that can also just put a pump and spaced passive vent on distro to empty it, which requires no access or insuls

They need to have access to an air alarm to change it to fill, no? That's the only time we set the override flag to true.

File: Content.Server/Atmos/Piping/Unary/Components/GasVentPumpComponent.cs (hunk 2 above)
Comment (MjrLandWhale) [reply Content.Server/Atmos/Piping/Unary/Components/GasVentPumpComponent.cs]: yeehaw

**Comment by deltanedas** [comment]
yes im saying this doesnt make sabotaging distro easier because its already trivial without air alarms

**Comment by MjrLandWhale** [comment]
> yes im saying this doesnt make sabotaging distro easier because its already trivial without air alarms

"passive vent".... Ahh reading comprehension, my nemesis.

**Comment by MjrLandWhale** [comment]
@EmoGarbage404 looks like it tagged you because I made some additions to one of the guidebook entries for atmos. The atmos 'fire & spacing' entry was pretty sparse to begin with, and this just gives some documentation of different options to fix spacings (including using this PR's new pressure lockout override).

**Review by Partmedia — APPROVED** [review]

**Comment by Partmedia** [comment]
Thanks for your contribution!

**Pull request merged**